module/VuFind/src/VuFind/ILS/Driver/SierraRest.php

```diff
@@ -1898,7 +1921,11 @@ protected function getApiUrlFromHierarchy(array $hierarchy): string
     {
         $url = $this->config['Catalog']['host'];
         foreach ($hierarchy as $value) {
-            $url .= '/' . urlencode($value);
+            if (is_array($value)) {
+                $url .= (str_contains($url, '?') ? '&' : '?') . http_build_query($value);
```

Comment:
The whole thing of having query parameters in the URL is not a recommended practice. You're right though, this could be used to create bogus url's, but I couldn't come up with anything that wouldn't get confusing for GET requests without breaking back-compatibility.

Reply:
If you think it's better to add an "additional GET parameters" (that would get merged with $params for GET requests), I could go with that.

Reply:
I think the additional GET parameters approach feels cleaner and would reduce the chances of invalid URL generation... but if it requires a significant amount of extra infrastructure to make it work (e.g. changing lots of method signatures and/or revising large amounts of code), I could live with this as-is since its scope of use is very limited.